### PR TITLE
read source from file

### PR DESCRIPTION
**Pull request by abhishek97**
Fixed : #8 

Used HTML5 FileReader API to read file to editor.

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -92,4 +92,19 @@ $(document).ready(function () {
         $(this).closest('li').addClass('active');
         init();
     });
+  
+  $('#uploadFile').click(function(e){
+    e.preventDefault();
+    $('#upload').click();
+  });
+  
+  var fileInput = document.getElementById('upload');
```
Comment (omerjerk) [inline scripts/main.js]: Before anything else, fix the indentation first.

Comment (abhishek97) [inline scripts/main.js]: @omerjerk  just fixed. :)

File: scripts/main.js [hunk 2]
```diff
@@ -92,4 +92,20 @@ $(document).ready(function () {
...
+  fileInput.addEventListener('change', function(e) {
+      var file = fileInput.files[0];
+      var reader = new FileReader();
+      reader.onload = function(e) { // closure to set read data to editor
+          ace.edit("editor").setValue(reader.result);
+      }
+      reader.readAsText(file);	
+});
```
Comment (omerjerk) [inline scripts/main.js]: This does not seem right.

Comment (abhishek97) [inline scripts/main.js]: @omerjerk  what exactly? , `readAsText` ? 
can you reference the line numbers?

Comment (omerjerk) [inline scripts/main.js]: Indent at line 109. :P

**Comment by abhishek97** [comment]
fixed indentation. :#

**Comment by omerjerk** [comment]
I'll review the actual code now. 😂

**Comment by hereisnaman** [comment]
@abhishek97 please also update lang variable as per the uploaded file extension, provided it is one of the supported lang.

**Comment by prateek27** [comment]
issue

There should be option to see the currently selected language. @abhishek97 Can you do it ?

**Comment by prateek27** [comment]
@hereisnaman  creating a new issue for this.

**Comment by abhishek97** [comment]
@prateek27  yeah, working on it.

**Pull request merged**